src/solidOpts.ts

```diff
@@ -21,7 +24,24 @@ let elementDeleteQueue: ElementNode[] = [];
 function flushDeleteQueue(): void {
   for (let el of elementDeleteQueue) {
     if (Number(el._queueDelete) < 0) {
-      el.destroy();
+      if (el.preserve) {
```

Comment:
You can preserve the element from getting deleted by setting `_queueDelete = 0` on creation, so there is no reason to even change how the queue works.

Reply:
no, it's only added to the delete queue if it's `undefined`, if you start with `0` instead of `undefined`, or any other number, it will never be added to the queue and never deleted.